src/controllers/injuryController.js

```diff
@@ -0,0 +1,221 @@
+const moment = require('moment');
+const mongoose = require('mongoose');
+const InjurySeverity = require('../models/Injury');
+const Project = require('../models/project');
+
+// Severity mapping for transition
+const severityMapping = {
+  Minor: 'Low',
+  Major: 'Serious',
+  Medium: 'Medium',
+  Low: 'Low',
+  Serious: 'Serious',
+};
+
+const getInjuries = async (req, res) => {
+  try {
+    const { projectId, startDate, endDate } = req.query;
+
+    const query = {};
+
+    // Build query based on filters
+    if (projectId && projectId !== 'all') {
+      if (!mongoose.Types.ObjectId.isValid(projectId)) {
+        return res.status(400).json({ message: 'Invalid project ID format' });
+      }
+      query.projectId = new mongoose.Types.ObjectId(projectId);
+    }
+
+    if (startDate || endDate) {
+      query.date = {};
+
+      if (startDate) {
+        const startDateTime = moment(startDate).startOf('day').toDate();
+        query.date.$gte = startDateTime;
+      }
+
+      if (endDate) {
+        const endDateTime = moment(endDate).endOf('day').toDate();
+        query.date.$lte = endDateTime;
+      }
+    }
+
+    const injuries = await InjurySeverity.find(query).sort({ date: 1 });
+
+    // If no injuries found, return empty data structure
+    if (!injuries || injuries.length === 0) {
+      return res.status(200).json({
+        months: [],
+        serious: [],
+        medium: [],
+        low: [],
+      });
+    }
+
+    // Group data by month and severity
+    const groupedData = {
+      months: [],
+      serious: [],
+      medium: [],
+      low: [],
+    };
+
+    const monthsSet = new Set();
+    injuries.forEach((injury) => {
+      const month = moment(injury.date).format('MMM'); // Only month without year
+      monthsSet.add(month);
+    });
+
+    const months = Array.from(monthsSet);
+    // Sort months in chronological order
+    const monthOrder = [
+      'Jan',
+      'Feb',
+      'Mar',
+      'Apr',
+      'May',
+      'Jun',
+      'Jul',
+      'Aug',
+      'Sep',
+      'Oct',
+      'Nov',
+      'Dec',
+    ];
+    const sortedMonths = months.sort((a, b) => monthOrder.indexOf(a) - monthOrder.indexOf(b));
+    groupedData.months = sortedMonths;
+
+    sortedMonths.forEach((month) => {
+      // Handle both old and new severity levels
+      const seriousCount = injuries
+        .filter((injury) => {
+          const mappedSeverity = severityMapping[injury.severity] || injury.severity;
+          return moment(injury.date).format('MMM') === month && mappedSeverity === 'Serious';
```

Comment:
As requested in the [Frontend PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3692#pullrequestreview-3030321476), to handle date ranges for different years, we can think of modifying this section of the code.